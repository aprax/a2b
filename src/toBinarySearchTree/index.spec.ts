import toBst from '.';

const testData = [
	['1'],
	['1', '2'],
	['1', '2', '3'],
	['1', '2', '3', '4'],
	['1', '2', '3', '4', '5'],
	['a', 'd', 'z']
]
for (const nodes of testData) {
	it(`should match snapshot for ${nodes.toString()}`, () => {
		const actual = Array(nodes.length);
		toBst(nodes, actual);
		expect(actual).toMatchSnapshot();
	})
}
