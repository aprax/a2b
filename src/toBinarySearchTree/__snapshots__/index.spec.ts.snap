// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for 1 1`] = `
Array [
  "1",
]
`;

exports[`should match snapshot for 1,2 1`] = `
Array [
  "1",
  ,
  "2",
]
`;

exports[`should match snapshot for 1,2,3 1`] = `
Array [
  "2",
  "1",
  "3",
]
`;

exports[`should match snapshot for 1,2,3,4 1`] = `
Array [
  "2",
  "1",
  "3",
  ,
  ,
  ,
  "4",
]
`;

exports[`should match snapshot for 1,2,3,4,5 1`] = `
Array [
  "3",
  "1",
  "4",
  ,
  "2",
  ,
  "5",
]
`;

exports[`should match snapshot for a,d,z 1`] = `
Array [
  "d",
  "a",
  "z",
]
`;
